# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/worklet/WorkletImpl.h
# Commit: 5278e40b80c3
# Full Hash: 5278e40b80c3c4aa642a24a1ff5a50247b9b3126
# Author: Tom Ritter <tom@mozilla.com>
# Date: 2022-11-28 15:41:50
# Regressor Bug: 1778510
# File Overlap Count: 1
# Description:
#   Bug 1778510: Add RTPCallerType to GlobalObject r=asuth
#   
#   This centralizes the logic in one place.
#   
#   In order to do this, we will need to check the principal
# ==============================================================================

diff -r dba220581d93 -r 5278e40b80c3 dom/worklet/WorkletImpl.h
--- a/dom/worklet/WorkletImpl.h	Mon Nov 28 04:21:26 2022 +0000
+++ b/dom/worklet/WorkletImpl.h	Mon Nov 28 04:21:26 2022 +0000
@@ -82,6 +82,7 @@
   const Maybe<nsID>& GetAgentClusterId() const { return mAgentClusterId; }
 
   bool IsSharedMemoryAllowed() const { return mSharedMemoryAllowed; }
+  bool IsSystemPrincipal() const { return mIsSystemPrincipal; }
   bool ShouldResistFingerprinting() const {
     return mShouldResistFingerprinting;
   }
@@ -106,6 +107,8 @@
   ipc::PrincipalInfo mPrincipalInfo;
   // Accessed on only worklet parent thread.
   nsCOMPtr<nsIPrincipal> mPrincipal;
+  // For off-main-thread checks
+  bool mIsSystemPrincipal;
 
   const WorkletLoadInfo mWorkletLoadInfo;
 
